### PR TITLE
Create authenticaiton system #45

### LINKED ISSUE #45: Add authentication method
More than likely this will need to be implemented using omniauth with the gina auth system

### PR DESCRIPTION
**Pull request by grimm**
Fixes #45

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -84,3 +84,9 @@ gem "font-awesome-rails"
 group :development do
   gem 'web-console', '~> 2.0'
 end
+
+gem 'omniauth'
+gem 'omniauth-github'
```
Comment (teknofire) [inline Gemfile]: shouldn't have the github in here

File: app/controllers/exports_controller.rb [hunk 2]
```diff
@@ -32,7 +32,7 @@ def create
     @export.sites = search.to_global_id.to_s
 
     respond_to do |format|
-      if @export.save
+      if @export.save and signed_in?
```
Comment (teknofire) [inline app/controllers/exports_controller.rb]: don't test for signed_in? user here, we'll need to look at setting up cancancan for doing this sort of thing

File: app/views/layouts/_navbar.html.haml [hunk 3]
```diff
@@ -25,9 +34,10 @@
           View sites
 
       .btn-group
-        %button.btn.btn-primary.navbar-btn{ data: { behavior: 'export', toggle: 'tooltip', placement: 'top', container: 'body' }, title: 'Click here to export data for visible sites' }
-          %i.fa.fa-cloud-download
-          Export data
+        - if signed_in?
```
Comment (teknofire) [inline app/views/layouts/_navbar.html.haml]: We also don't want to hide this button if a user is not logged in, eventually we want the system to take them through the process of logging as part of the export

**Comment by grimm** [comment]
Ready for review.

**Comment by teknofire** [comment]
The one we don't need is the omniauth-github gem

**Comment by teknofire** [comment]
:+1:

**Pull request merged**